ext/hal/nxp/imx/devices/MCIMX7D/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+zephyr_include_directories(.)
+
+zephyr_sources(
+ system_MCIMX7D_M4.c
+ startup_MCIMX7D_M4.S
```

Comment:
I don't think we need to build either of these. The zephyr cortex-m port already does all or most everything in startup_MCIMX7D_M4.S. The zephyr mpu driver does a lot of what's in system_MCIMX7D_M4.c, so you might just need to grab the cache init stuff and copy it to soc.c. You can do a follow up patch later to enable the existing mpu driver.